Engine/ScriptAnalyzer.cs

```diff
@@ -876,39 +876,37 @@ private List<string> GetValidCustomRulePaths(string[] customizedRulePath, PathIn
 
         private IEnumerable<T> GetInterfaceImplementationsFromAssembly<T>(string ruleDllPath) where T : class
         {
-            var fileName = Path.GetFileNameWithoutExtension(ruleDllPath);
+            string fileName = Path.GetFileNameWithoutExtension(ruleDllPath);
             var assemblyName = new AssemblyName(fileName);
             outputWriter.WriteVerbose(string.Format("Loading Assembly:{0}", assemblyName.FullName));
 
-            var dll = Assembly.Load(assemblyName);
-            var rules = new List<T>();
+            Assembly dll = Assembly.Load(assemblyName);
+
             if (dll == null)
             {
                 outputWriter.WriteVerbose(string.Format("Cannot load {0}", ruleDllPath));
-                return rules;
+                return Enumerable.Empty<T>();
             }
-            foreach (var type in dll.ExportedTypes)
+
+            var rules = new List<T>();
+            foreach (Type type in dll.ExportedTypes)
```

Comment:
DRY:
```suggestion
            foreach (var type in dll.ExportedTypes)
```

Reply:
I don't think there's repetition here; the type of `ExportedTypes` is not inferable on that line. In fact, since it's property on `System.Reflection.Assembly`, it's not inferable anywhere in the program.